client/my-sites/checkout/checkout-thank-you/transfer-pending/index.tsx

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+import * as React from 'react';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import { useDispatch, useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import getAtomicTransfer from 'calypso/state/selectors/get-atomic-transfer';
+import { errorNotice } from 'calypso/state/notices/actions';
+import { transferStates } from 'calypso/state/atomic-transfer/constants';
+import { hideMasterbar, showMasterbar } from 'calypso/state/ui/masterbar-visibility/actions';
+import { useInterval } from '../../../../lib/interval/use-interval';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+// Total time to perform "loading"
+const DURATION_IN_MS = 60000;
+
+// Props type
+interface Props {
+	siteId: number;
+	orderId: number;
+}
+
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+const TransferPending: React.FunctionComponent< Props > = ( props ) => {
+	const { __ } = useI18n();
+	const { siteId, orderId } = props;
+	const dispatch = useDispatch();
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const transfer = useSelector( ( state ) => getAtomicTransfer( state, siteId ) );
+
+	const steps = React.useRef< string[] >( [
+		__( 'Setting up your site' ),
+		__( 'Upgrading infrastructure' ),
+		__( 'Preparing WooCommerce' ),
+	] );
+
+	const totalSteps = steps.current.length;
+
+	const [ currentStep, setCurrentStep ] = React.useState( 0 );
+
+	/**
+	 * Completion progress: 0 <= progress <= 1
+	 */
+	const progress = ( currentStep + 1 ) / totalSteps;
+	const isComplete = progress >= 1;
+
+	useInterval(
+		() => setCurrentStep( ( s ) => s + 1 ),
+		// Enable the interval when progress is incomplete
+		isComplete ? null : DURATION_IN_MS / totalSteps
+	);
+
+	// Force animated progress bar to start at 0
+	const [ hasStarted, setHasStarted ] = React.useState( false );
+	React.useEffect( () => {
+		const id = setTimeout( () => setHasStarted( true ), 750 );
+		return () => clearTimeout( id );
+	}, [] );
```

Comment:
The first opinion that came to my mind about creating this state is it could be discarded. We are creating two states to deal with the progress of the bar. `currentStep` and `hasStarted`. Do you think it's avoidable?

<img src="https://user-images.githubusercontent.com/77539/127906515-c60381f0-5f94-49f0-9a68-10ec39c89b9b.png" width="400px" />
Maybe playing with the currentState, adding an initial step, or something like that.
Please consider this as a suggestion.

Reply:
Thanks! I tried out some ways of getting around the two states (checking that `progress` is 0, removing the conditional and just showing `progress`, adding an initial step) but it seems like the logic depends on it. The `hasStarted` state prevents the first `0` step from progressing too early, and at the same time we're relying on `steps` and `currentStep` to have the right step numbers, including `0`. It would definitely be interesting to refactor the component and see if we could avoid it!